week1/tuesday/Assignment_1_2_3/main.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+**  This program prompts user for the amount of students, and for each, asks
+**  name, age and grade. It then calculates and prints the average grade
+**  and name of the highest graded student.
+*/
+
+#include <iostream>
+#include <limits>
+#include <vector>
+
+struct Student
+{
+    std::string name {};
+    int         age {};
+    int         grade {};
+};
+
+void clear_cin()
+{
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+}
+
+std::string get_name()
+{
+    std::string name {};
+    std::cout << "Enter the student's name: ";
+    std::getline(std::cin >> std::ws, name);
+    return name;
+}
+
+int get_age(std::string name)
+{
+
+    int age { 0 };
+    while (true)
+    {
+        std::cout << "Enter " << name << "'s age: ";
+        std::cin >> age;
+        if (!std::cin)
+            clear_cin();
+        if (age > 0 && age < 125)
+            return age;
+        std::cout << "Invalid age! Try again... :)\n";
+    }
+}
+
+int get_grade(std::string name)
+{
+    int grade { 0 };
+    while (true)
+    {
+        std::cout << "Enter " << name << "'s grade (1-5): ";
+        std::cin >> grade;
+        if (!std::cin)
+            clear_cin();
+        if (grade > 0 && grade <= 5)
+            return grade;
+        std::cout << "Invalid grade! Try again... :)\n";
+    }
+}
+
+/*
+**  Populates the vector of type struct Student and returns the populated vector
+*/
+std::vector<Student> create_database(int amount)
+{
+    std::vector<Student> students;
+
+    for (size_t i = 0; i < static_cast<size_t>(amount); ++i)
+    {
+        students.push_back(Student());
+        students[i].name = get_name();
+        students[i].age = get_age(students[i].name);
+        students[i].grade = get_grade(students[i].name);
+        std::cout << '\n';
+    }
+    students.shrink_to_fit();
+    return students;
+}
```

Comment:
Could create Student with all bells and whistles and then add it to vector. 
This abstract away from understanding what is being modified.